detection_rules/rule.py

```diff
@@ -592,6 +601,23 @@ def get_required_fields(self, index: str) -> List[dict]:
         if validator is not None:
             return validator.get_required_fields(index or [])
 
+    def validate_subclasses(self) -> None:
+        """Check that the query rule subclasses have validate and call method."""
+
+        def validate_recursive(obj: Any) -> None:
```

Comment:
Nit. Since this function is designed to be called with the internal `self` being passed on line 620, could we have the type hint be `QueryRuleData` object since it should always be this in the current implementation?

Reply:
Good idea. It's not always `QueryRuleData`, it could be any rule dataclass defined in `rule.py`, so type hinting for `dataclass` may be our only option. I expanded on this a bit in the comment above.